storage/blockdevice/source/SFDP.cpp

```diff
@@ -291,13 +291,96 @@ int sfdp_parse_sector_map_table(Callback<int(bd_addr_t, sfdp_cmd_addr_size_t, ui
         return -1;
     }
 
-    // Currently we support only Single Map Descriptor
-    if (!((smptbl_buff[0] & 0x3) == 0x03) && (smptbl_buff[1] == 0x0)) {
-        tr_error("Sector Map: Supporting Only Single Map Descriptor (not map commands)");
+    // Loop through all descriptors: configuration command descriptor and sector map descriptors.
+    // Command descriptors are optional. If any exist, they always precede sector map descriptors.
+    uint8_t *table = smptbl_buff.get();
+    uint8_t *descriptor = table;
+    const size_t min_descriptor_size = 8; // two DWORDs
+
+    // Part 1: configuration detection command descriptors (optional).
+    // If any exist, run their commands to determine the active configuration's ID.
+    // Otherwise the configuration ID is 0x00.
+    uint8_t active_config_id = 0x00;
+    // If any configuration detection command descriptors exist, the table starts with one.
+    if (!(descriptor[0] & 0x02)) {
+        while (descriptor + min_descriptor_size <= table + sfdp_info.smptbl.size) { // table boundary
+            if (descriptor[0] & 0x02) {
+                tr_error("Sector Map: Expecting a configuration detection command descriptor");
+                return -1;
+            }
+
+            uint8_t instruction = descriptor[1];
+            uint8_t dummy_cycles = descriptor[2] & 0x0F;
+            auto addr_size = static_cast<sfdp_cmd_addr_size_t>(descriptor[2] >> 6);
+            uint8_t mask = descriptor[3];
+            uint32_t cmd_addr;
+            memcpy(&cmd_addr, &descriptor[4], sizeof(cmd_addr)); // bytes 4-7 of the descriptor
```

Comment:
Is this `memcpy` a performance optimisation? Can we use copy-assignment or `std::copy` to do this in a less error-prone way?

Reply:
The left hand side `cmd_addr` is a 32-bit unsigned integer, but the right hand side `descriptor` is an array of 8-bit unsigned integers. This kind of assignment is semantically only doable with `memcpy`, and it's the safest when alignment is not guaranteed.